doc/development/powershell.md

```diff
@@ -0,0 +1,179 @@
+This page contains guidelines for developing or updating powershell scripts used by the Azure SDK engineering system.
+
+Table of Contents
+=================
+
+* [TLDR](#tldr)
+* [Structure](#structure)
+* [Functionality](#functionality)
+* [Style](#style)
+* [Testing](#testing)
+   * [Unit Testing](#unit-testing)
+      * [Running Pester Tests](#running-pester-tests)
+   * [Local/Pipeline Functional Testing](#localpipeline-functional-testing)
+
+## TLDR
+
+_(i.e. too long, didn't read)_
+
+**DO**
+
+- Try to write scripts that can be run locally
+- Write unit tests for your scripts. See [Unit Testing](#unit-testing).
+- Handle exit codes from external programs
+- Use `Set-StrictMode -Version 4` where possible
+- Define your data structures as much as possible via composition, or unit test or comment examples.
+- Use an editor extension or `Invoke-ScriptAnalyzer` to analyze code.
+
+**DON'T**
+
+- Write medium-large scripts inline to azure pipelines yaml (unless absolutely necessary)
+- Write code without putting logic into functions. Scripts grow in size and complexity over time, and if they start out un-organized, they tend to stay un-organized.
+
+## Structure
+
+When structuring a powershell cmdlet or collection of scripts, it is important to think about how easy that code will be to re-use and to test (primarily unit tests and/or running locally). 
```

Comment:
For scripts we should also include CmdBinding so we get the PS common parameters defined as well.